platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpoint.java

```diff
@@ -159,73 +155,65 @@ public Response create(MultipartBody multipartBody, @Context UriInfo requestUriI
+      if (inputStream.available() == 0) {
         inputStream.reset();
       }
+
+      String filename = getFileName(attachment.getContentDisposition());
+
+      if (!filename.endsWith(JAR_EXT) && !filename.endsWith(KAR_EXT)) {
+        LOGGER.debug("Wrong file type: {}", FilenameUtils.getExtension(filename));
+        Response.ResponseBuilder responseBuilder = Response.serverError();
+        responseBuilder.status(HttpStatus.UNSUPPORTED_MEDIA_TYPE_415);
+        return null;
+      }
+
+      File uploadDir = new File(defaultFileLocation);
+      if (!uploadDir.exists() && uploadDir.mkdirs()) {
+        LOGGER.info("Unable to make directory {}", uploadDir.getAbsolutePath());
+      }
+
+      File newFile = new File(uploadDir, filename);
+      FileUtils.copyInputStreamToFile(inputStream, newFile);
+      return newFile;
+
     } catch (IOException e) {
-      LOGGER.debug("IOException reading stream from file attachment in multipart body", e);
-      IOUtils.closeQuietly(inputStream);
+      LOGGER.debug("Unable to write file", e);
+      return null;
     }
+  }
 
-    if (filename.endsWith(JAR_EXT) || filename.endsWith(KAR_EXT)) {
-      if (inputStream != null) {
-        try {
-          File uploadDir = new File(defaultFileLocation);
-          if (!uploadDir.exists()) {
-            if (uploadDir.mkdirs()) {
-              LOGGER.info("Unable to make directory {}", uploadDir.getAbsolutePath());
-            }
-          }
-
-          newFile = new File(uploadDir, filename);
-
-          FileUtils.copyInputStreamToFile(inputStream, newFile);
-
-        } catch (IOException e) {
-          LOGGER.debug("Unable to write file.", e);
-          newFile = null;
-        } finally {
-          IOUtils.closeQuietly(inputStream);
-        }
-      } else {
-        LOGGER.debug("No file attachment found");
-      }
-    } else {
-      LOGGER.debug("Wrong file type.");
-      Response.ResponseBuilder responseBuilder = Response.serverError();
-      responseBuilder.status(HttpStatus.UNSUPPORTED_MEDIA_TYPE_415);
-      IOUtils.closeQuietly(inputStream);
+  private String getFileName(ContentDisposition contentDisposition) {
+    if (contentDisposition == null) {
+      LOGGER.debug(FILENAME_NOT_FOUND_MSG, DEFAULT_FILE_NAME);
+      return DEFAULT_FILE_NAME;
     }
-    return newFile;
+
+    String filename = contentDisposition.getParameter(FILENAME_CONTENT_DISPOSITION_PARAMETER_NAME);
+    if (StringUtils.isEmpty(filename)) {
+      LOGGER.debug(FILENAME_NOT_FOUND_MSG, DEFAULT_FILE_NAME);
+      return DEFAULT_FILE_NAME;
+    }
+
+    filename = FilenameUtils.getName(filename);
+    LOGGER.debug("Filename: {}", filename);
+    return filename;
   }
 
   /**
    * Setter method for DEFAULT_FILE_LOCATION for testing purposes
    *
    * @param fileLocation the desired fileLocation
    */
-  void setDefaultFileLocation(String fileLocation) {
+  static void setDefaultFileLocation(String fileLocation) {
```

Comment:
✏️  It doesn't look like the javadoc is giving any valuable information. Maybe change it to an `@VisibleForTesting`?